pkg/resource/atomic.go

```diff
@@ -15,7 +15,8 @@ type CreateFn func(id string) proto.Message
 type GetFn func() (item proto.Message, err error)
 
 // ChangeFn is called to apply changes to the new proto.Message.
-type ChangeFn func(old, new proto.Message) error
+// due to handling new = nil, it returns any modifications made to new
```

Comment:
Can you rewrite this line, it doesn't really stand up as an independent comment. If you instead describe the expected values/nil-ness of the args and the return value that should cover what you're trying to say.

Reply:
Again, the comment reads from your current pov - i.e. trying to fix these initial value panics.

```go
// ChangeFn is called to apply changes to a proto.Message.
// old and new are the current message and a message to change.
// If old is nil, there is no current message.
// If new is nil, then the resource was unable to allocate a message of the correct type to apply changes to.
// The implementation is free to ignore both arguments but must return a message containing the desired changes.
```

How does that sound?

Reading my comment back it might make sense to rename the `new` arg to `dst` instead, I think that might make things clearer in the docs.